src/main.rs

```diff
@@ -274,6 +274,18 @@ fn main() {
         std::process::exit(exitcode::CONFIG);
     });
 
+    let mut config_watcher =
+        match topology::config::watcher::config_watcher(&config, &opts.config_path) {
+            Ok(watcher) => watcher,
+            Err(error) => {
+                error!(
+                    message = "Configuration file watcher creation failed",
+                    ?error
+                );
+                std::process::exit(exitcode::CONFIG);
```

Comment:
I would assume this exit code means, a config error not a failure to open a notify channel?

Reply:
No, you are right. 
At this stage config file was confirmed to exist, so this error will happen, in most cases, if something happened with the file. So exitcode::IOERR could be a better fit?

Reply:
_probably_, I think that makes sense!